lib/iris/cube.py

```diff
@@ -2302,7 +2306,8 @@ def _as_list_of_coords(self, names_or_coords):
         Convert a name, coord, or list of names/coords to a list of coords.
         """
         # If not iterable, convert to list of a single item
-        if not hasattr(names_or_coords, '__iter__'):
+        if (not hasattr(names_or_coords, '__iter__') or
+                isinstance(names_or_coords, str)):
```

Comment:
This method takes a `str` name, `Coord` or list/tuple/iterable of the same. This check makes it a list if it's not, so you can't just check if it's `str`.